src/hooks/useSubmitTransaction.tsx

```diff
@@ -40,12 +40,12 @@ const useSubmitTransaction = ({
   const submitTransaction = async () => {
     try {
       setSubmitting(true)
-      const amountToShow =
+      const finalAmount =
         mode === ModeOptions.payment
-          ? +amount + totalFeeUsd
+          ? (+amount).toFixed(decimals || 6)
           : feeDeduct
-            ? +amount
-            : +amount + totalFeeUsd
```

Comment:
Yah this looks correct now.
* In `payment` mode the seller must receive the full amount
* The amount in `/submit` should be the amount the target receives, so this would be less the fees in the `deduct` case